### PR TITLE
Java version, Gradle and Gradle Plugin updates

### PR DESCRIPTION
**Pull request by DinoMarlir**
- Java release: 16 -> 17
- Kotlin JVM: 1.7.20 -> 1.8.10

**NEW VERSION**
1.0.0 -> 1.0.1

### CODE REVIEW

**Comment by DinoMarlir** [comment]
@mooziii should I remove the OptIn annontation over the main function?

**Comment by mooziii** [comment]
nein ich release jetzt kein update nur wegen version bumps vergiss es

**Comment by iswearimmrtobi** [comment]
würd mir stinken

**Comment by DinoMarlir** [comment]
@iswearimmrtobi why?

**Comment by mooziii** [comment]
das ist der größte contributions farmen moment

**Comment by DinoMarlir** [comment]
@mooziii i wanted to help you to keep your versions up to date, there is no reason to react agressive.

**Comment by iswearimmrtobi** [comment]
wenn ich eine Flasche so fest zudrehe, dass ich sie nicht mehr öffnen kann. Bin ich dann stark oder schwach?

**Comment by DinoMarlir** [comment]
@iswearimmrtobi I don't understand what you mean...

**Comment by DinoMarlir** [comment]
@mooz where is the problem with my contribution? What can I improve?

**Comment by mooziii** [comment]
1. I don't see a reason to update this anymore as it does what it should and if there is no new feature, bug fix or whatever I won't release an update for it / merge the PR as I'm not going to touch this ever again
2. Why would you bump the java version? There is absolutely no reason for this upgrade as it only makes some systems incompatible

**Comment by mooziii** [comment]
3. On Windows 11 systems the issue seems to be fixed and I'm expecting Microsoft to fix this in Windows 10 too (maybe 23H1? Idk)

**Comment by DinoMarlir** [comment]
@mooziii I changed the Java version in the `build.gradle.kts` file because Java 16 is not a Java LTS version and support ended on 6 Mar 2021 and no longer receives updates.

**Comment by mooziii** [comment]
yeah but what has this to do with the jdk version the jar was built with? Actually I wanted to use java 8 but due to some kotlin features only being available for java 16 I choose to use 16 back then

**Comment by DinoMarlir** [comment]
@mooziii I can change the java version back to 16 if you want.

**Comment by DinoMarlir** [comment]
What type of feature do you want to have in 'snipping tool fix'? Maybe a autofix that automatically kills the task?

**Comment by mooziii** [comment]
I already thought of an autokill feature but the problem there is that you won't be able to take screenshots at all if the tasks gets killed instantly

**Comment by DinoMarlir** [comment]
Maybe we could do that with a delay.

**Comment by DinoMarlir** [comment]
Oh the task starts when you press the hotkey, right?

**Comment by mooziii** [comment]
> Oh the task starts when you press the hotkey, right?
yes, this feature would only be great if you can detect if the screenshot was taken or not

**Comment by DinoMarlir** [comment]
Isn't there a way to see if the menu has been closed (e.g. if something goes to the clipboard)?

File: build.gradle.kts [hunk 1]
```diff
@@ -16,11 +16,11 @@ dependencies {
 
 tasks {
     compileJava {
-        options.release.set(16)
+        options.release.set(17)
```
Comment (mooziii) [inline build.gradle.kts:19]: Change this back to 16 (maybe try 11?)

File: build.gradle.kts [hunk 2]
```diff
@@ -1,10 +1,10 @@
 plugins {
-    kotlin("jvm") version "1.7.20"
-    id("com.github.johnrengelman.shadow") version "7.1.2"
+    kotlin("jvm") version "1.8.10"
+    id("com.github.johnrengelman.shadow") version "8.1.0"
 }
 
 group = "me.obsilabor"
-version = "1.0.0"
+version = "1.0.1"
```
Comment (mooziii) [inline build.gradle.kts:7]: Please remove this version bump if no new features will be added

**Review by mooziii — CHANGES_REQUESTED** [review]

File: build.gradle.kts (hunk 1 above)
Comment (DinoMarlir) [reply build.gradle.kts:19]: 👍

Comment (DinoMarlir) [reply build.gradle.kts]: I changed it to `11`

File: src/main/kotlin/me/obsilabor/snippingtoolfix/SnippingToolFix.kt [hunk 3]
```diff
@@ -7,6 +7,7 @@ import java.net.URI
 import java.net.URL
 import javax.imageio.ImageIO
 import kotlin.jvm.optionals.getOrNull
+import kotlin.streams.toList
```
Comment (mooziii) [inline src/main/kotlin/me/obsilabor/snippingtoolfix/SnippingToolFix.kt:10]: ?

Comment (DinoMarlir) [reply src/main/kotlin/me/obsilabor/snippingtoolfix/SnippingToolFix.kt:10]: The toList function has to be imported in Java 11.

**Comment by mooziii** [comment]
so Kollege Schnürschuh wie siehts aus

**Comment by DinoMarlir** [comment]
What type of feature do you want to have?

**Pull request closed without merging**